addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -128,7 +128,9 @@ def _create_invoice(self, order, so_line, amount):
 
         if order.fiscal_position_id:
             invoice_vals['fiscal_position_id'] = order.fiscal_position_id.id
-        invoice = self.env['account.move'].sudo().create(invoice_vals).with_user(self.env.uid)
+
+        invoice = self.env['account.move'].with_company(order.company_id).with_user(self.env.uid)\
+            .sudo().create(invoice_vals)
```

Comment:
```suggestion
        invoice_sudo = self.env['account.move'].with_company(order.company_id).sudo().create(
            invoice_vals)
        invoice = invoice_sudo.with_user(self.env.uid)
```

Please do not alter the order of calls without knowing what they do.  `with_user` resets the sudo and should be done AFTER the creation done in sudo to make sure the returned record is not sudoed for the subsequent operations.

Reply:
Hello Victor,
Thanks for your comment. I'm new to Odoo and I will take into account your comment for my future PRs.

I made the modification you suggested.